src/lib/ActionPrototype.js

```diff
@@ -1,29 +1,57 @@
-var fetcher = require('../lib/jsonFetcher');
-var AppDispatcher = require('../lib/ep-dispatcher');
-var KeyMirror = require('keymirror');
+import fetcher from '../lib/jsonFetcher'
+import AppDispatcher  from '../lib/ep-dispatcher'
+import KeyMirror from 'keymirror'
+
+//TODOS: 
+//--turn into a class. 
+//--make it a factory function like we do with store prototype. 
 
 var ActionPrototype = {
   specialTypes: KeyMirror({
     GOT_MSG: null
   }),
+  originalActions: {},
+  stubForTests: function() {
+    this.recievedData = {}
+    let actions = Object.getOwnPropertyNames(this).filter( (fn)=> {return typeof this[fn] === 'function' } )
+    actions.forEach( (action)=> {
+      if (['fireApi', 'stubForTests', 'stubedAction', 'saveAction', 'restore'].indexOf(action) < 0) {
+        this.saveAction(action)
+        this[action]=(data)=> {this.stubedAction(action, data)}
```

Comment:
@BJK Missing spaces around the equals sign, before the arrow, and inside the braces.